pkg/rehearse/jobs.go

```diff
@@ -817,29 +813,34 @@ func (e *Executor) ExecuteJobs() (bool, error) {
 	return waitSuccess, err
 }
 
-func (e *Executor) waitForJobs(jobs sets.String, selector string) (bool, error) {
+func (e *Executor) waitForJobs(jobs sets.String, selector ctrlruntimeclient.ListOption) (bool, error) {
 	if len(jobs) == 0 {
 		return true, nil
 	}
 	success := true
-	for {
-		w, err := e.pjclient.Watch(metav1.ListOptions{LabelSelector: selector})
-		if err != nil {
-			return false, fmt.Errorf("failed to create watch for ProwJobs: %v", err)
-		}
-		defer w.Stop()
-		for event := range w.ResultChan() {
-			prowJob, ok := event.Object.(*pjapi.ProwJob)
-			if !ok {
-				return false, fmt.Errorf("received a %T from watch", event.Object)
+	var listErrors []error
+	if err := e.pollFunc(10*time.Second, 4*time.Hour, func() (bool, error) {
+		result := &pjapi.ProwJobList{}
+		// Don't bail out just because one LIST failed
+		if err := e.pjclient.List(context.Background(), result, selector, ctrlruntimeclient.InNamespace(e.namespace)); err != nil {
+			if len(listErrors) > 2 {
+				return false, utilerrors.NewAggregate(append(listErrors, err, errors.New("encountered more than three subsequent errors trying to list")))
```

Comment:
I think the "more than three" message is misleading, because we actually bail out on three.

Reply:
Updated